sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java

```diff
@@ -104,40 +126,77 @@ public RecordNetworkCallPolicy(RecordedData recordedData) {
-            return response.bodyAsByteArray().map(bytes -> {
+            return response.bodyAsByteArray().switchIfEmpty(Mono.just(new byte[0])).map(bytes -> {
+                if (bytes.length == 0) {
+                    return responseData;
+                }
+
                 String content;
-                try (GZIPInputStream gis = new GZIPInputStream(new ByteArrayInputStream(bytes));
-                     ByteArrayOutputStream output = new ByteArrayOutputStream()) {
-                    byte[] buffer = new byte[DEFAULT_BUFFER_LENGTH];
-                    int position = 0;
-                    int bytesRead = gis.read(buffer, position, buffer.length);
-
-                    while (bytesRead != -1) {
-                        output.write(buffer, 0, bytesRead);
-                        position += bytesRead;
-                        bytesRead = gis.read(buffer, position, buffer.length);
+                if ("gzip".equals(response.headerValue("Content-Encoding"))) {
+                    try (GZIPInputStream gis = new GZIPInputStream(new ByteArrayInputStream(bytes));
+                         ByteArrayOutputStream output = new ByteArrayOutputStream()) {
+                        byte[] buffer = new byte[DEFAULT_BUFFER_LENGTH];
+                        int position = 0;
+                        int bytesRead = gis.read(buffer, position, buffer.length);
+
+                        while (bytesRead != -1) {
+                            output.write(buffer, 0, bytesRead);
+                            position += bytesRead;
+                            bytesRead = gis.read(buffer, position, buffer.length);
+                        }
+
+                        content = new String(output.toByteArray(), StandardCharsets.UTF_8);
+                    } catch (IOException e) {
+                        throw Exceptions.propagate(e);
                     }
-
-                    content = new String(output.toByteArray(), StandardCharsets.UTF_8);
-                } catch (IOException e) {
-                    throw Exceptions.propagate(e);
+                } else {
+                    content = new String(bytes, StandardCharsets.UTF_8);
                 }
 
-                responseData.remove("content-encoding");
-                responseData.put("content-length", Integer.toString(content.length()));
+                responseData.remove("Content-Encoding");
+                responseData.put("Content-Length", Integer.toString(content.length()));
 
-                responseData.put("body", content);
+                responseData.put("Body", content);
                 return responseData;
             });
         }
     }
+
+    private String redactUserDelegationKey(String content) {
+        if (!content.contains("UserDelegationKey")) {
+            return content;
+        }
+
+        content = redactionReplacement(content, DELEGATIONKEY_KEY_PATTERN.matcher(content), Base64.getEncoder().encodeToString("REDACTED".getBytes(StandardCharsets.UTF_8)));
```

Comment:
Since you have the helper method, it is better to simplify the second param to string

Reply:
I'd rather handle the encoding here as the other redactions don't require an encoding.

Reply:
I mean you can have the signature like
redactionReplacement(String content, String regex, String replacement)
Not a big fan of method call in signature. Hard to log and debug. The third param  makes sense to me as some key might use other way to encode.